### PR TITLE
ensure private field is set by the client

### LINKED ISSUE #465: Private not being set on projects by default
Projects being created don't have private field set to nil, which means they fail the show / index visible_scope test.

- [x] A projects private field should be configurable from the front end. 
- [ ] It should default to true.
- [x] It should be a required param. 
- [x] Serialise the private field in the response.
- [x] update docs accordingly.

### PR DESCRIPTION
**Pull request by camallen**
closes #465 - I think the client should explicitly set false / true with the front end defaulting to true (private).

### CODE REVIEW

File: app/schemas/project_create_schema.rb [hunk 1]
```diff
@@ -19,6 +19,10 @@ class ProjectCreateSchema < JsonSchema
       description "Two character ISO 638 language code, optionally include two character ISO 3166-1 alpha-2 country code seperated by a hyphen for specific locale. ie 'en', 'zh-tw', 'es_MX'"
     end
 
+    property "private" do
+      type "boolean"
```
Comment (edpaget) [inline app/schemas/project_create_schema.rb]: Can we make this field required in the schema if we're going to require it to create a project? See https://github.com/camallen/Panoptes/blob/projects_private_fix/app/schemas/subject_set_create_schema.rb#L5

**Comment by edpaget** [comment]
Does this resolve #465 and #466?

**Comment by camallen** [comment]
This resolves #465 only. #466 seemed like non-intuitive behaviour to me.

**Comment by edpaget** [comment]
What do you mean sorry?

On Fri, Jan 30, 2015 at 2:15 PM, Campbell Allen <notifications@github.com>
wrote:

> This resolves #465 <https://github.com/zooniverse/Panoptes/issues/465>
> only. #466 <https://github.com/zooniverse/Panoptes/issues/466> seemed
> like non-intuitive behaviour to me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zooniverse/Panoptes/pull/467#issuecomment-72263464>.
>



-- 
Ed Paget

**Comment by camallen** [comment]
I manually set the first 3 projects that @brian-c made after rebuilding the DB to private: true. But they were then visible to non-authenticated user (when private is nil no projects are listed as per https://github.com/zooniverse/Panoptes-Front-End/issues/88).

It seems counter intuitive to me based on the public_scope and the roles defined here: https://github.com/zooniverse/Panoptes/blob/c7afefee4bc54ebd14803e2c9e1ca630141315e5/app/models/project.rb#L25

Thoughts?

**Comment by edpaget** [comment]
That makes sense. I'll write a test case and work on it.

File: app/schemas/project_create_schema.rb (hunk 1 above)
Comment (camallen) [inline app/schemas/project_create_schema.rb]: done

**Pull request merged**